crossfire/__init__.py

```diff
@@ -43,17 +43,18 @@ def occurrences(
     )
 
 
-def _flatten_dict(data, parent_key, sep="_"):
-    for dict in data:
-        for k, v in dict.items():
-            new_key = parent_key + sep + k if parent_key else k
-            yield new_key, v
-
-
 def flatten(data, nested_columns=None):
     if nested_columns is None:
         nested_columns = NESTED_COLUMNS
-    elif nested_columns not in (NESTED_COLUMNS):
+    nested_columns = set(nested_columns)
+    if not nested_columns.issubset(NESTED_COLUMNS):
         raise ValueError(f"Invalid `nested_columns` value: {nested_columns}")
     if isinstance(data, list):
-        return _flatten_dict(data, nested_columns)
+        for dict in data:
+            for nested_column in nested_columns:
+                if nested_column in dict:
```

Comment:
I am not sure if this makes sense, but I added this validation just in case a potential nested column doesn't exist in the dict.

Reply:
We a talking about two sets:
1. all keys of a dictionary
2. `nested_columns` input by the user

Nesting one `for` loop inside the other makes this operation too costly when compared to getting the intersection of two sets: `keys = set(dict.keys()) & nested_columns`.

Also, avoid calling the new variable `dict` in `for dic in data` — you are overwriting [native `dict`](https://docs.python.org/3.12/library/stdtypes.html#dict).

Finally, can we expect all dictionaries in `data` to have the same keys? If so, we can extract the dictionary keys out of any `for` loop.

Reply:
> Right?

Right.

> Using listcomprehension?

List comprehension is another notation of a `for` loop. So, no. But you skip the important part: _ can we expect all dictionaries in data to have the same keys?_

Reply:
> can we expect all dictionaries in data to have the same keys?

I think so. But I believe that we can asume it will be filled with value...
A little bit lost on how to sove your enigma :)

Reply:
OK. And we are doing that and also getting the insersection with `nested_columns`:
`keys = set(item.keys()) & nested_columns`
But, then I need to iterate over it to flatten the columns, like here:
```python
        for item in data:
            keys = set(item.keys()) & nested_columns
            for key in keys:
                for k, v in item.get(key).items():
                    new_key = (
                        key + "_" + k if nested_columns else k
                    )
                    return new_key, v
```
Seems that I still using too many `for` loops...
I believe that yout will answer asking for to code to be commited, so... commiting rigth now.

Reply:
Take a look in this code you shared:

```python
  for item in data:
        keys = set(item.keys()) & nested_columns
```

If:
* `data` is [what I suggested](https://github.com/FelipeSBarros/crossfire/pull/100#discussion_r1487976161) (i.e. if we assume, as [you said](https://github.com/FelipeSBarros/crossfire/pull/100#discussion_r1487939303) dictionary keys are always the same)
* and `nested_columns = {"a"}`

How different will `keys` be in each iteration of the `for` loop?

Reply:
> How different will keys be in each iteration of the for loop?

I am not sure if I should suppose that. But in the case a data comes with no `contextInfo`, for instance. But as you asked, "is it possible to have different keys between `data` items?". Idon't think so.
Actually, I am now considering this as an unecessary point. Those nested columns should always exists. Bot perhaps some times can come with `None` values.
Nevertheless, your point is why I am geeting this keys intersection for every `data` item if they must have the same keys. Right? 
But, let suppose we will keep this key identification using intersection... you suggest to get it from the first item instead of getting keys for each item?

> How different will keys be in each iteration of the for loop?

Yes, I believe  they won't be different.

Reply:
> Yes, I believe they won't be different.

So, why not define `keys` just once, instead of redefining it in every iteration of the loop?

```python
keys = set(data[0].keys()) & nested_columns  
for item in data:
       pass
```

Reply:
NOw I got you. First I though you was suggesting a way to use less `for`loops...